src/forms/TextFieldRule/TextFieldRule.jsx

```diff
@@ -0,0 +1,96 @@
+import PropTypes from 'prop-types';
+import { useState, useEffect, useMemo } from 'preact/hooks';
+import TextField from '../TextField/TextField';
+import Text from '../../typography/Text/Text';
+import CSS from './styles.css';
+
+const TextFieldRule = ({ className, dataTestId, name, readOnly, placeholder, errorLabel, value, type, rule, onChange, onBlur, startFocused, ...props }) => {
+	const [currentValue, setCurrentValue] = useState('');
+
+	useEffect(() => {
+		if(!value) { return; }
+		setCurrentValue(value);
+	}, []);
+
+	const handleOnChange = eventValue => {
+		if (currentValue !== eventValue) {
+			setCurrentValue(eventValue);
+		}
+		if (onChange) {
+			onChange(eventValue);
+		}
+	};
+
+	const handleOnBlur = event => {
+		const {
+			target: { eventValue },
+		} = event;
+		if (currentValue !== eventValue) {
+			setCurrentValue(eventValue);
+		}
+		if (onBlur) {
+			onBlur(event);
+		}
+	};
+
+	const result = useMemo(() => {
+		const labeledClass = (currentValue) ? CSS.alignTop : '';
+		return (
+			<div className={CSS.wrapper}>
+				<TextField 
+					dataTestId={dataTestId} 
+					placeholder={placeholder} 
+					className={className} 
+					type={type} 
+					name={name} 
+					value={currentValue} 
+					errorLabel={errorLabel} 
+					onChange={handleOnChange} 
+					onBlur={handleOnBlur} 
+					maxLength={rule} 
+					startFocused={startFocused} 
+					{...props} 
+				/>
+				<Text 
+					dataTestId={`${dataTestId}-rule`} 
+					color={`ash`} 
```

Comment:
(nit) please simplify this line because it doesn't need the template and backtick syntax:

```js
color="ash"
```